dapp/src/helpers/publicAttestation.ts

```diff
@@ -0,0 +1,50 @@
+export interface Twitter {
+    // Allows for case switching against variant.
+    type: 'twitter'
+    handle: string,
+    pubkey: string,
+} 
+
+export interface Discord {
+    // Allows for case switching against variant.
+    type: 'discord'
+    handle: string,
+    pubkey: string
+}
+
+export interface Dns {
+    // Allows for case switching against variant.
+    type: 'dns'
+    domain: string,
+    pubkey: string
+}
+
+export type variant = Twitter | Discord | Dns;
+
+const fnMap = {
+    'discord': (discord: Discord) => {
+        return `I am attesting that this discord handle ${discord.handle} is linked to the Tezos account ${discord.pubkey} for @tzprofiles\n\n`;
+    },
+    'twitter': (twitter: Twitter) => {
+        return `I am attesting that this twitter handle ${twitter.handle} is linked to the Tezos account ${twitter.pubkey} for @tzprofiles\n\n`;
+    },
+    'dns': (dns: Dns) => {
+        return `${dns.domain} is linked to ${dns.pubkey}`
+    }
+}
+
+export function tzprofileChecker(variant: variant): string {
+    let t = variant.type;
+    switch (variant.type) {
```

Comment:
Enums don't work like in Rust, you still need a default? Is that what the following comment is about?

Reply:
Neither Enums nor Sum Types are supported outside of TS (rather, in JS proper). Your checks can only be at compile time, so at runtime, folks can still pass bad arguments. 

I want to unify the two side's implementation at some point, preferably in Rust where the type support is better, and you just hit serialization errors if you pass a bad arg.